### PR TITLE
Fix not working select of existing excerpt

### LINKED ISSUE #185: Fix not working select on select existing excerpt the second time
## To reproduce:
1. Select an existing excerpt from the list by clicking its name in the list
2. Modify the excerpt on the map &rarr; the application will change to the 'new excerpt' form
3. Change back to the list of existing excerpt
4. Try to re-choose the previously chosen excerpt by clicking on its name in the list

### expected:
* The excerpt on the map snaps back to the extend of the chosen excerpt.

### observed:
* Nothing happens.

Estimated duration: ~1d

### PR DESCRIPTION
**Pull request by wasabideveloper**
Implementation of fix #185 (Merge will close #185 ).

* Tested in IE11, FF and Chrome and it works well

To be reviewed by:
- [x] @das-g 
- [x] @hixi

### CODE REVIEW

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js [hunk 1]
```diff
@@ -74,7 +84,7 @@
 
         this._setLocationFilterFromExcerptID = function(ID) {
             var that = this;
-            this.selectedExcerptGeoJson = L.geoJson.ajax("/api/bounding_geometry_from_excerpt/"+ID+"/").on('data:loaded', function(){
+            this.selectedExcerptGeoJson = L.geoJson.ajax("/api/bounding_geometry_from_excerpt/"+ID+"/?format=json").on('data:loaded', function(){
```
Comment (das-g) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js]: @hixi Should clients need to specify `format=json`? I'd expect this to be the default if the client doesn't specify any preferred content types. (Or _does_ `L.geoJson.ajax` specify some HTML content types as preferred, thus making it necessary to override that with an URL parameter?)

**Comment by das-g** [comment]
Has this been tested on multiple browsers? (I have no idea how portable or browser-specific the used JavaScript features are.)

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js (hunk 1 above)
Comment (hixi) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js]: No, it shouldn't. If it is indeed needed right now, I'd consider this a bug on the osamaxx rest api client. 

We probably just haven't set the settings:

```python
REST_FRAMEWORK = {
    'DEFAULT_RENDERER_CLASSES': (
        'rest_framework.renderers.JSONRenderer',
    ),
    'DEFAULT_PARSER_CLASSES': (
        'rest_framework.parsers.JSONParser',
    )
}
```

See: [REST Framework Settings](http://www.django-rest-framework.org/api-guide/settings/#settings)

Comment (wasabideveloper) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js]: It was not working without specifying `format=json` so I hat to use that. I'm open for a other solution...

Comment (hixi) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js]: See comment above ;-)

Comment (wasabideveloper) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js]: :heavy_check_mark:

**Comment by wasabideveloper** [comment]
I'm testing on IE. But there we have some strange behaviour anyway. I will try to improve it.

**Comment by wasabideveloper** [comment]
@das-g and @hixi please take a look again. I replaced `formPartsSwitcher.dispatchEvent(new Event('change'));` by my own method (&rarr;extract callback to onFormPartsSwitcherChange) to support IE and enhance decoupling.

**Comment by hixi** [comment]
:+1:

**Pull request merged**